src/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Hll.hpp

```diff
@@ -0,0 +1,185 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/NewtonianEuler/BoundaryCorrections/BoundaryCorrection.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace NewtonianEuler::BoundaryCorrections {
+/*!
+ * \brief An HLL (Harten-Lax-van Leer) Riemann solver for NewtonianEuler system
+ *
+ * Let \f$U\f$ be the evolved variable, \f$F^i\f$ the flux, and \f$n_i\f$ be the
+ * outward directed unit normal to the interface. Denoting \f$F := n_i F^i\f$,
+ * the HLL boundary correction is \cite Harten1983
+ *
+ * \f{align*}
+ * G_\text{HLL} = \frac{\lambda_\text{max} F_\text{int} +
+ * \lambda_\text{min} F_\text{ext}}{\lambda_\text{max} - \lambda_\text{min}}
+ * - \frac{\lambda_\text{min}\lambda_\text{max}}{\lambda_\text{max} -
+ *   \lambda_\text{min}} \left(U_\text{int} - U_\text{ext}\right)
+ * \f}
+ *
+ * where "int" and "ext" stand for interior and exterior.
+ * \f$\lambda_\text{min}\f$ and \f$\lambda_\text{max}\f$ are defined as
+ *
+ * \f{align*}
+ * \lambda_\text{min} &=
+ * \text{min}\left(\lambda^{-}_\text{int},-\lambda^{+}_\text{ext}, 0\right) \\
+ * \lambda_\text{max} &=
+ * \text{max}\left(\lambda^{+}_\text{int},-\lambda^{-}_\text{ext}, 0\right)
+ * \f}
+ *
+ * where \f$\lambda^{+}\f$ (\f$\lambda^{-}\f$) is the characteristic speed
+ * largest in outgoing (ingoing) direction. Note the minus signs in front of
+ * \f$\lambda_\text{ext}\f$s, which is because outgoing speed for exterior
+ * corresponds to ingoing speed for interior, and vice versa. By the same token,
+ * the positive sign in front of the \f$F_{\text{ext}}\f$ in \f$G_\text{HLL}\f$
+ * is necessary because the outward directed normal of the neighboring element
+ * has the opposite sign, i.e. \f$n_i^{\text{ext}}=-n_i^{\text{int}}\f$.
```

Comment:
"By the same token, the positive sign in front of the \f$F_{\text{ext}}\f$ in \f$G_\text{HLL}\f$ is necessary because" -> "Similarly, the \f$F_{\text{ext}}\f$ term in \f$G_\text{HLL}\f$ has a positive sign because"